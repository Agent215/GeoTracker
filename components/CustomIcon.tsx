import Drought from '../assets/Icons/drought.svg';
import DustStorm from '../assets/Icons/dustStorm.svg';
import Earthquakes from '../assets/Icons/earthQuake.svg';
import TempExtremes from '../assets/Icons/extremeTemp.svg';
import Floods from '../assets/Icons/flood.svg';
import SeaLakeIce from '../assets/Icons/SeaLakeIce.svg';
import RockMudSlide from '../assets/Icons/rockMudSlide.svg';
import Manmade from '../assets/Icons/manmade.svg';
import SevereStorm from '../assets/Icons/severeStorm.svg';
import Volcano from '../assets/Icons/volcano.svg'
import Wildfire from '../assets/Icons/wildfire.svg'
import SnowFall from '../assets/Icons/snowFall.svg'

import React from 'react';


/* This function creates the correct custom icon using an SVG file,
then returns this component. It takes two props, category and size to display. */
const DisasterIcon = (props) => {

    /* add categories to ENUM object, not sure how to extract the width and height here,
    there is probably a better way to do this so we dont have to repeat the height and width everytime.*/
    const categorySVG = {
        ["drought"]: <Drought width={props.size} height={props.size} />,
        ["dustHaze"]: <DustStorm width={props.size} height={props.size} />,
        ["earthquakes"]: <Earthquakes width={props.size} height={props.size} />,
        ["tempExtremes"]: <TempExtremes width={props.size} height={props.size} />,
        ["floods"]: <Floods width={props.size} height={props.size} />,
        ["seaLakeIce"]: <SeaLakeIce width={props.size} height={props.size} />,
        ["landslides"]: <RockMudSlide width={props.size} height={props.size} />,
        ["manmade"]: <Manmade width={props.size} height={props.size} />,
        ["severeStorms"]: <SevereStorm width={props.size} height={props.size} />,
        ["snow"]: <SnowFall width={props.size} height={props.size} />,
        ["volcanoes"]: <Volcano width={props.size} height={props.size} />,
        ["wildfires"]: <Wildfire width={props.size} height={props.size} />
    }

    const eventIcon = categorySVG[props.event.category];
    console.log("This is " + props.event.category);
    if (!eventIcon) {
        console.log("CustomIcon.js/EventIcon() - No event of type" + props.event + " found");
        return null
    };

    return (eventIcon);

}

export default DisasterIcon;