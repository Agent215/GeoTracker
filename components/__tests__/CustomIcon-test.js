import * as React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux'
import configureStore from 'redux-mock-store'
import CustomIcon from '../../components/CustomIcon';
import * as enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import MockedNavigator from '../MockedNavigator'


const mockedNavigate = jest.fn();

jest.mock('@react-navigation/native', () => {
  return {
    ...jest.requireActual('@react-navigation/native'),
    useNavigation: () => ({
      navigate: mockedNavigate,
    }),
  };
});

enzyme.configure({ adapter: new Adapter() });
// mock of props to pass disaster card
const size= 22;
const event =   { title: 'Lava', id: 5, category: 'clouds', currentLat: 40.059813, currentLong: -75.190574, sourceLink: 'https://eonet.sci.gsfc.nasa.gov/', date: "" }


describe('Testing CustomIcon Component', () => {
  const initialState = { output: "null" }
  const mockStore = configureStore()
  let store, wrapper
  /*
  Check CustomIcon against snapshot. run jest -u to update snapshot
  */
  it(`CustomIcon renders correctly`, () => {
    store = mockStore(initialState)

    const comp = <Provider store={store}><CustomIcon size={size} event={event} /></Provider>
    const tree = renderer.create(comp).toJSON();

    expect(tree).toMatchSnapshot();
  });

 
})
