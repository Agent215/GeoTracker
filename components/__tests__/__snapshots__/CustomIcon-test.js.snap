// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing CustomIcon Component CustomIcon renders correctly 1`] = `null`;
